### PR TITLE
Convert Compaction to use React.js

### PR DESCRIPTION
**Pull request by garrensmith**
This converts all backbone.js views to React and Flux.

### CODE REVIEW

File: app/addons/compaction/tests/storesSpec.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  'testUtils'
+], function (FauxtonAPI, Stores, testUtils) {
+  var assert = testUtils.assert;
+  var store = Stores.compactionStore;
+
+  describe('Compaction Store', function () {
+
+
+  });
+});
```
Comment (robertkowalski) [inline app/addons/compaction/tests/storesSpec.js]: can we delete this?

File: app/addons/compaction/views.js [hunk 2]
```diff
@@ -19,73 +19,10 @@ define([
 ],
 function (app, FauxtonAPI, Compaction) {
 
-  Compaction.Layout = FauxtonAPI.View.extend({
-    template: 'addons/compaction/templates/layout',
+  // This is no longer used. We need to come up with a solution to allow to compact a specific view with CouchDB 2.0
+  // Once that is supported we can convert this to React.
```
Comment (robertkowalski) [inline app/addons/compaction/views.js]: Compaction just works on 15986 - but it works.

Why do we comment all this old code?

**Comment by garrensmith** [comment]
@robertkowalski I've removed the test

**Comment by sebastianrothbucher** [comment]
+1 on Robert's comment - can we quite possibly just get rid of views.js alltogether? Also I asked myself (during databases implementation) whether I should keep all IDs or adjust the travis test. I decided for the earlier. Anyway, the "am I running on backdoor" and nightwatch are the only two obstacles I can make out. Apart from that, it looks just OK to me (and it works for me also ;-) )

File: app/addons/compaction/components.react.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      var btnText = 'Run';
+
+      if (this.props.isCompacting) {
+        btnText = 'Compacting...';
+      }
+
+      return (
+        <div className="row-fluid">
+          <div className="span12 compaction-option">
+            <h3> Compact Database </h3>
```
Comment (robertkowalski) [inline app/addons/compaction/components.react.jsx]: pls remove the sourrounding space.

it leads to the invariant issues we had with react

File: app/addons/compaction/components.react.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    render: function () {
+      var btnText = 'Run';
+
+      if (this.props.isCleaningView) {
+        btnText = 'Cleaning Views...';
+      }
+      return (
+        <div className="row-fluid">
+          <div className="span12 compaction-option">
+            <h3> Cleanup Views </h3>
```
Comment (robertkowalski) [inline app/addons/compaction/components.react.jsx]: possible invariant

**Comment by robertkowalski** [comment]
+1, merge if tests are green! :)

**Comment by garrensmith** [comment]
Merged.

**Comment by michellephung** [comment]
this is cool garren!

**Pull request closed without merging**